### PR TITLE
adding exercises for week4 and week5

### PR DESCRIPTION
**Pull request by pythonwa**
Hi Professor, 
I am submitting these files for week 4 and week 5 exercises. 

Thanks Professor

### CODE REVIEW

File: students/tonylee/mailroom_part2.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    choice = 0
+    while(choice != 4):
+        choice = input('''Please choose from a menu of 3 actions:\n
+                    1 - Send a Thank You
+                    2 - Create a Report
+                    3 - Send letters to everyone
+                    4 - Quit\n''')
+        choice = int(choice)
+        if choice >=1 and choice < 4:
+            OptionSelect[choice]()
```
Comment (charlesrobison) [inline students/tonylee/mailroom_part2.py]: Works well!

File: students/tonylee/mailroom_part3.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+    if name not in donationHistory.keys():
+        donationHistory[name] = []
+
+    donationAmount = input("Please enter the donation amount: ")
+    try:
+        donationHistory[name].append(float(donationAmount))
+        print(f'Thank you {name}, for your donation of {donationAmount}!')
+    except ValueError:
+        print("Not a valid donation amount. Pleae re-enter a valid donation amount.")
```
Comment (charlesrobison) [inline students/tonylee/mailroom_part3.py]: Typo - "Please re-enter..."

File: students/tonylee/mailroom_part3.py [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+    choice = 0
+    while(choice != 4):
+        choice = input('''Please choose from a menu of 3 actions:\n
+                    1 - Send a Thank You
+                    2 - Create a Report
+                    3 - Send letters to everyone
+                    4 - Quit\n''')
+        choice = int(choice)
+        if choice >=1 and choice < 4:
```
Comment (charlesrobison) [inline students/tonylee/mailroom_part3.py]: Recommend adding an exception for a non numeric input.

File: students/tonylee/mailroom_part2.py [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+def thankyou():
+    name = input("Please enter the full name: ")
+    while name == "list":
+        for donorName in donationHistory.keys():
+            print(donorName)
+        name = input("Please enter the full name: ")
+
+    if name not in donationHistory.keys():
+        donationHistory[name] = []
```
Comment (natasha-aleksandrova) [inline students/tonylee/mailroom_part2.py]: per pep8 use snake_case style for var names -> donation_history

File: students/tonylee/mailroom_part2.py [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+    for name, donationAmount in donationHistory.items():
+        print('{:20}|{:15}|{:10}|{:12}'.format(name, sum(donationAmount), len(donationAmount), sum(donationAmount)/len(donationAmount)))
+
+def send_letter():
+    for name, donationAmount in donationHistory.items():
+        with open(name+'.txt', 'w') as letterFile:
+            letterFile.write('Thank you {}, for your donation of {}!'.format(name, sum(donationAmount)))
+
+if __name__ == '__main__':
+    OptionSelect = {
```
Comment (natasha-aleksandrova) [inline students/tonylee/mailroom_part2.py]: snake_case (along with other vars in your code) -> `option_select`

**Pull request merged**